ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@rossberg-chromium Rather than revert, could we do `br0`/br0_if`?  Then we keep the syntactic arity match (which is what postorder needs) and then this PR would switch to `check_type` of an `expr` and the result-dropping works.